pkg/controller/hacks/hacks.go

```diff
@@ -19,3 +23,37 @@ func ReduceLikelihoodOfRepeatedReconciliation(ctx context.Context) {
 	log.Info("Waiting 2 seconds to give the cache a chance to sync after updating resource")
 	time.Sleep(2 * time.Second)
 }
+
+// WorkAroundTypeObjectProblemInCRDSchemas works around the problem where OpenShift 3.11 doesn't like "type: object"
+// in CRD OpenAPI schemas. This function removes all occurrences from the schema.
+func WorkAroundTypeObjectProblemInCRDSchemas(ctx context.Context, err error, cl client.Client, crd *unstructured.Unstructured) error {
+	log := common.LogFromContext(ctx)
+	if err != nil && strings.Contains(err.Error(), "must only have \"properties\", \"required\" or \"description\" at the root if the status subresource is enabled") {
+		log.Info("The API server rejected the CRD. Removing type:object fields from the CRD schema and trying again.")
+
+		schema, found, err := unstructured.NestedFieldNoCopy(crd.UnstructuredContent(), "spec", "validation", "openAPIV3Schema")
```

Comment:
Should we be handling the `!found && err==nil` case here? IIUC that should never happen, but if it does for whatever reason, we'll not output an error but also not create the CRD

Reply:
In that case, the original error is returned and is logged.

I've already changed this code in a different PR I'm about to submit. I've had to move the `cl.Create` call out of this function, as I need to call `cl.Update` in one different case.

Reply:
Oh, you're right, I misread that